voxblox_ros/src/visualize_tsdf.cc

```diff
@@ -0,0 +1,205 @@
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <pcl/conversions.h>
+#include <pcl/io/ply_io.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl_msgs/PolygonMesh.h>
+#include <pcl_ros/point_cloud.h>
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+
+#include <voxblox/core/tsdf_map.h>
+#include <voxblox/io/layer_io.h>
+#include <voxblox/io/mesh_ply.h>
+#include <voxblox/mesh/mesh_integrator.h>
+
+#include "voxblox_ros/mesh_pcl.h"
+#include "voxblox_ros/mesh_vis.h"
+#include "voxblox_ros/ptcloud_vis.h"
+
+DEFINE_string(tsdf_proto_path, "", "Path to the serialized TSDF grid.");
+
+DEFINE_double(tsdf_surface_distance_threshold_factor, 0.75,
+              "This factor multiplied with the voxel size determines the "
+              "maximum distance to the surface for voxels to be considered "
+              "near-surface voxels.");
+
+DEFINE_string(tsdf_world_frame, "world",
+              "Name of the world frame. This frame will be used to publish all "
+              "the pointclouds and meshes.");
+
+DEFINE_string(tsdf_mesh_color_mode, "color",
+              "Color mode for the TSDF mesh extraction, options: "
+              "[color, height, normals, lambert, gray]");
+
+DEFINE_string(tsdf_voxel_ply_output_path, "",
+              "If specified, the pointcloud representing all voxels will be "
+              "saved to a ply file.");
+
+namespace voxblox {
+class SimpleTsdfVisualizer {
+ public:
+  SimpleTsdfVisualizer(const ros::NodeHandle& nh,
+                       const ros::NodeHandle& nh_private)
+      : nh_(nh), nh_private_(nh_private) {
+    VLOG(1) << "\tSetting up ROS publishers...";
+
+    surface_pointcloud_pub_ =
+        nh_private_.advertise<pcl::PointCloud<pcl::PointXYZRGB>>(
+            "tsdf_voxels_near_surface", 1, true);
+
+    tsdf_pointcloud_pub_ =
+        nh_private_.advertise<pcl::PointCloud<pcl::PointXYZI>>(
+            "all_tsdf_voxels", 1, true);
+
+    mesh_pub_ = nh_private_.advertise<voxblox_msgs::Mesh>("mesh", 1, true);
+
+    mesh_pointcloud_pub_ =
+        nh_private_.advertise<pcl::PointCloud<pcl::PointXYZRGB>>(
+            "mesh_as_pointcloud", 1, true);
+
+    mesh_pcl_mesh_pub_ =
+        nh_private_.advertise<pcl_msgs::PolygonMesh>("mesh_pcl", 1, true);
+
+    ros::spinOnce();
+  }
+
+  void run(const Layer<TsdfVoxel>& tsdf_layer);
+
+ private:
+  ros::NodeHandle nh_;
+  ros::NodeHandle nh_private_;
+
+  ros::Publisher surface_pointcloud_pub_;
+  ros::Publisher tsdf_pointcloud_pub_;
+  ros::Publisher mesh_pub_;
+  ros::Publisher mesh_pointcloud_pub_;
+  ros::Publisher mesh_pcl_mesh_pub_;
+};
+
+void SimpleTsdfVisualizer::run(const Layer<TsdfVoxel>& tsdf_layer) {
+  LOG(INFO) << "\nTSDF Layer info:\n"
+            << "\tVoxel size:\t\t " << tsdf_layer.voxel_size() << "\n"
+            << "\t# Voxels per side:\t " << tsdf_layer.voxels_per_side() << "\n"
+            << "\tMemory size:\t\t " << tsdf_layer.getMemorySize() * 1e-6
```

Comment:
I don't think this is quite right, probably doesn't matter, but I think what you wanted to do is `tsdf_layer.getMemorySize() * pow(2,20)` (or using a bit shift
```c++
size_t memory_size = tsdf_layer.getMemorySize();
memory_size = memory_size << 20;
```
)